kubernetes/structures_rbac.go

```diff
@@ -89,7 +89,9 @@ func expandClusterRoleAggregationRule(in []interface{}) *api.AggregationRule {
 
 	if v, ok := m["cluster_role_selectors"].([]interface{}); ok && len(v) > 0 {
 		crs := make([]metav1.LabelSelector, 0)
```

Comment:
Small possible optimization here. The length of slice `crs` will be equal to the length of slice `v`. What would you say if we pass `len(v)` as length and capacity when making `crs` and then within the for loop use indexes to assign values instead of `append`?

Reply:
That was my initial intention, but I looked at the rest of the code and copied the pattern most used to keep it consistent. I'll make this change.

Reply:
RE acceptance test - Is it multi pass or do we need tests for each case? Off the top of my head there are a minimum of 5 test scenarios to prove that the aggregation is working correctly.